packages/core/src/lib/store.types.ts

```diff
@@ -60,6 +62,10 @@ export interface WalletSelectorState {
    * Returns ID-s of 5 recently signed in wallets.
    */
   recentlySignedInWallets: Array<string>;
+  /**
+   * The list of contracts when SignInMulti is supported.
+   */
+  contracts: MultiContractState | null;
```

Comment:
We need to handle cases where:

1. A dapp might call `signIn()` for a contract that was previously signed into using `signInMulti()`.  For example, let's say that a dapp uses 3 contracts regularly, and their access key allowance expires for just _one_ of those three contracts.  The dapp author does not need to request the user re-signIn to all 3 contracts using `signInMulti()` -- they would just request `signIn()` for the _one contract_ whose allowance has been used up.
2. A dapp might call `signInMulti()` with an array that includes a contract that they had _previously_ signed in to using `signIn()`

If we always store results from `signInMulti()` in `contracts` and results from `signIn()` in `contract`, we will end up with duplicate entries in these keys when a mixture of calls is used by the dapp developer -- we need to have one-and-only-one entry for a given contract across these two fields (`contract` and `contracts`) to make sure everything is sane and deterministic.

Reply:
This PR should solve these issues: #847